### PR TITLE
Split up TOML tests based on pass/fail

### PR DESCRIPTION
**Pull request by ben-albrecht**
Added a `Passing.chpl` test for the passing tests of the BurntSushi suite. (#7104)
This will remove any passing futures from testing.

As tests pass, they should be migrated over to `Passing.execopts`

Also removed a unintended copy of the future (`TestToml.future`)

**Testing**
- [x] `start_test test/library/packages/TOML/BurntSushi`

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@benharsh or @lydia-duncan - care to review?

File: test/library/packages/TOML/BurntSushi/TomlTest.future [hunk 1]
```diff
@@ -1,4 +1,8 @@
 feature request: Improve TOML library
 Issue #7104
 
-Test taken from BurntSushi/toml-test
+Test taken from BurntSushi test suite:
+
+    https://github.com/BurntSushi/toml-test
+
+Passing tests are listed in Passing.execopts
```
Comment (lydia-duncan) [inline test/library/packages/TOML/BurntSushi/TomlTest.future]: Would you mind calling out the desire to merge Passing and this future together into a single test when the rest pass, either here or in the linked issue?

Comment (ben-albrecht) [reply test/library/packages/TOML/BurntSushi/TomlTest.future]: Yes!

Comment (ben-albrecht) [reply test/library/packages/TOML/BurntSushi/TomlTest.future]: I mean no, I would not mind.

Comment (lydia-duncan) [reply test/library/packages/TOML/BurntSushi/TomlTest.future]: lol, I always have that trouble with answering "would you mind" requests, yet I use them myself anyways XD

**Pull request merged**